Charts: Storybook intro page and readme update

adamwoodnz opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Closes [CHARTS-62](https://linear.app/a8c/issue/CHARTS-62/charts-hosted-storybook)

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Adds an intro page to the Charts Storybook which will be the live public landing page at https://automattic.github.io/jetpack-storybook/?path=/docs/js-packages-charts
* Updates the shared Storybook config, with custom ordering for the Charts package to ensure that the Intro page comes first
* Updates the readme to reference the live Storybook and adds more details, consistent with intro page

<img width="2800" height="2376" alt="localhost_50240__path=_docs_js-packages-charts-introduction--docs" src="https://github.com/user-attachments/assets/6b648231-c09d-4ef7-a979-6e3b7edf202f" />

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Run Storybook
* Go to http://localhost:50240/?path=/docs/js-packages-charts
* Ensure the Introduction page is loaded
* Read content and assess accuracy, legibility, maintainability
* Check links
* Observe the order of the stories in the sidebar and compare with https://automattic.github.io/jetpack-storybook: only the JS Packages story ordering should have changed, with Charts being in the correct alphabetical position and the Introduction story first
* View the updated readme https://github.com/Automattic/jetpack/blob/c91917857526ec737dc44fec7f416d9360f7cd85/projects/js-packages/charts/README.md
* Read content and assess accuracy, legibility, maintainability
* Check links (note that the link to the live Storybook will be broken until this PR is deployed)

adamwoodnz committed:
Add charts storybook docs intro

adamwoodnz committed:
Add changelog

adamwoodnz committed:
Update readme
- update intro and link to storybook intro
- add installation and contributing details

kangzj reviewed (dismissed):
Changes and new doc look good to me! All links work too 🚀

adamwoodnz committed:
Update chart documentation title and enhance interactivity description

adamwoodnz committed:
Add story sorting options in Storybook preview configuration

- Updated the order for charts stories, including ensuring that Intro comes first

adamwoodnz committed:
Add Storybook changelog

adamwoodnz committed:
Update Storybook link in charts README

kangzj reviewed (approved):
🚀
